### PR TITLE
[SPARK-35366][SQL] Avoid using deprecated `buildForBatch` and `buildForStreaming`

### PR DESCRIPTION
**Pull request by linhongliu-db**
### What changes were proposed in this pull request?
Currently, in DSv2, we are still using the deprecated `buildForBatch` and `buildForStreaming`.
This PR implements the `build`, `toBatch`, `toStreaming` interfaces to replace the deprecated ones.

### Why are the changes needed?
Code refactor

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
exsting UT

### CODE REVIEW

**Comment by linhongliu-db** [comment]
cc @cloud-fan

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWrite.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package org.apache.spark.sql.kafka010
+
+import java.{util => ju}
+
+import org.apache.spark.sql.connector.write.{BatchWrite, Write}
+import org.apache.spark.sql.connector.write.streaming.StreamingWrite
+import org.apache.spark.sql.types.StructType
+
+case class KafkaWrite(
```
Comment (cloud-fan) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWrite.scala:25]: shall we implement `Write.description`?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWrite.scala [hunk 2]
```diff
@@ -29,24 +29,25 @@ import org.apache.spark.internal.io.FileCommitProtocol
...
 import org.apache.spark.sql.types.{DataType, StructType}
 import org.apache.spark.sql.util.SchemaUtils
 import org.apache.spark.util.SerializableConfiguration
 
-abstract class FileWriteBuilder(
-    paths: Seq[String],
-    formatName: String,
-    supportsDataType: DataType => Boolean,
-    info: LogicalWriteInfo) extends WriteBuilder {
+trait FileWrite extends Write {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWrite.scala:40]: ditto

**Review by HeartSaVioR — COMMENTED** [review]
Looks OK to me. Would be nice to address existing review comments as well.

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWrite.scala (hunk 1 above)
Comment (linhongliu-db) [reply external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWrite.scala:25]: sure

Comment (linhongliu-db) [reply external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWrite.scala:25]: resolved

**Comment by linhongliu-db** [comment]
@cloud-fan @HeartSaVioR, thanks for reviewing. comments are addressed.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**